x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/search/SearchApplication.java

```diff
@@ -231,6 +229,10 @@ public long updatedAtMillis() {
         return searchApplicationTemplate;
     }
 
+    public SearchApplicationTemplate searchApplicationTemplateOrDefault() {
+        return searchApplicationTemplate != null ? searchApplicationTemplate : SearchApplicationTemplate.DEFAULT_TEMPLATE;
```

Comment:
Related to the above, probably the getter method for `searchApplicationTemplate` should do this logic, so it always retrieves the default template when none is stored.

Reply:
I thought it was better to add an `orDefault` method so we could validate what was sent in with the raw command.

Reply:
> we could validate what was sent in with the raw command.

Sorry, what do you mean by "raw command"?

We have the hasStoredTemplate() method for checking whether the actual template is stored. Returning the default template on the getter would avoid having to distinguish cases on the GET request...

Reply:
Yeah, now that we're returning the default template in the response I'm OK cleaning this up.